### PR TITLE
Instancing: Clean ups, simpler Skin Cluster(s) updates

### PR DESCRIPTION
**Pull request by samcake**
Cleaning up left over comments and typos, simplify the code to update Matrices in the model classes and removed renderITemsIdMap because not needed, chasing the problem of the physics objects transform

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Can someone from High Fidelity approve this PR?

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/22017/HighFidelity-Beta-PR16377-695b9cd.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/22017/HighFidelity-Beta-Interface-PR16377-695b9cd.exe

**Comment by hifi-gustavo** [comment]
Android build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22017/HighFidelity-Beta-PR16377-695b9cd.apk).  Quest build is available [here](https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/22017/HighFidelity-Quest-Beta-PR16377-695b9cd.apk)

File: libraries/render-utils/src/Model.cpp [hunk 1]
```diff
@@ -740,9 +739,9 @@ bool Model::replaceScriptableModelMeshPart(scriptable::ScriptableModelBasePointe
         render::Transaction transaction;
         for (int i = 0; i < (int) _modelMeshRenderItemIDs.size(); i++) {
             auto itemID = _modelMeshRenderItemIDs[i];
-            auto shape = _modelMeshRenderItemShapes[i];
+            auto& shape = _shapeStates[i];
             // TODO: check to see if .partIndex matches too
```
Comment (sabrina-shanman) [inline libraries/render-utils/src/Model.cpp]: What's up with this comment?

File: libraries/render-utils/src/Model.h [hunk 2]
```diff
@@ -339,18 +349,13 @@ class Model : public QObject, public std::enable_shared_from_this<Model>, public
...
         std::vector<TransformDualQuaternion> clusterDualQuaternions;
         std::vector<glm::mat4> clusterMatrices;
-    };
 
-    const MeshState& getMeshState(int index) { return _meshStates.at(index); }
-
-    class ShapeState {
-    public:
-        glm::mat4 _rootFromJointTransform;
+        uint32_t getNumClusters() const { return (uint32_t) std::max(clusterMatrices.size(), clusterMatrices.size()); }
```
Comment (sabrina-shanman) [inline libraries/render-utils/src/Model.h]: Is it intentional to call clusterMatrices.size() twice?

File: libraries/render-utils/src/Model.cpp [hunk 3]
```diff
@@ -1415,33 +1411,30 @@ void Model::updateClusterMatrices() {
...
-                auto jointPose = _rig.getJointPose(cluster.jointIndex);
+                auto jointPose = _rig.getJointPose(cbmov.jointIndex);
                 Transform jointTransform(jointPose.rot(), jointPose.scale(), jointPose.trans());
                 Transform clusterTransform;
                 Transform::mult(clusterTransform, jointTransform, cbmov.inverseBindTransform);
                 state.clusterDualQuaternions[clusterIndex] = Model::TransformDualQuaternion(clusterTransform);
-            } else {
-                auto jointMatrix = _rig.getJointTransform(cluster.jointIndex);
+            }
+            else {
```
Comment (sabrina-shanman) [inline libraries/render-utils/src/Model.cpp]: Minor code style issue

File: libraries/render-utils/src/Model.cpp (hunk 1 above)
Comment (samcake) [reply libraries/render-utils/src/Model.cpp]: this was there and i din't change that

File: libraries/render-utils/src/Model.h (hunk 2 above)
Comment (samcake) [reply libraries/render-utils/src/Model.h]: no it s not good catch!

**Pull request merged**